### ISSUE TITLE
1.5.X Find Element By Name Removed

### ISSUE THREAD

**Issue by JordanGS** [appium/appium#6641]
## Feedback

I don't think this should have been removed, it was fine being depreciated. Not every app has id or accessibility id's and not everyone can easily add those in, nor do companies have time to dedicate to rewriting a working application. Most operate on a basis of "if it ain't broke, don't fix it". From a developer standpoint, I am now forced to choose between upgrading my appium version for new features or upgrading ALL of my tests and thousands of variables to not use name. :( Removal of functions should be limited to major revisions. Appium version 2.0 for example and not to minor revisions, in this case Appium 1.5.

## The Problem

Massive errors when trying to install Appium 1.4.16
-You can't even downgrade appium if you're unhappy with version 1.5.X which is due t0 node compatibility, if you downgrade node to 0.12 and then try. You can but you're unable to run appium because of "Appium will not work if used or installed with sudo" which it wasn't.

**Comment by jmolliver** [comment]
I've found that .elementById works exactly the same way as .elementByName. I only had to do a replace and did not have to change any of the underlying code. 
You can see they updated the sample file with it here. Let me know if that works for you.
appium/sample-code@a820e49#diff-9bd16273653a34774dbf75d755146c30

**Comment by SrinivasanTarget** [comment]
@triager please close

**Comment by JordanGS** [comment]
@jmolliver thanks but that didn't work. I ended up using xpath to search for `@text` on android for `@name` on iOS.
